velox/functions/lib/Re2Functions.cpp

```diff
@@ -378,7 +379,28 @@ class LikeConstantPattern final : public VectorFunction {
  public:
   LikeConstantPattern(StringView pattern, std::optional<char> escapeChar)
       : re_(toStringPiece(likePatternToRe2(pattern, escapeChar, validPattern_)),
-            RE2::Quiet) {}
+            RE2::Quiet) {
+    pattern_ = pattern;
+    fixedPattern_ = false;
+    prefixPattern_ = false;
+    suffixPattern_ = false;
+
+    if (escapeChar) {
```

Comment:
It would be nice to document this limitation in the PR description? Also, what would it take to remove this limitation and allow for efficient matching of `LIKE 'a_b' ESCAPE '_'`?

Reply:
Sure, will update the PR description. I am not sure how this limitation can be overcome and need to look into it.